src/compiler/emitter.ts

```diff
@@ -7086,7 +7125,22 @@ var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, Promi
             }
 
             function emitDetachedComments(node: TextRange) {
-                let leadingComments = getLeadingCommentRanges(currentSourceFile.text, node.pos);
+                let leadingComments: CommentRange[];
+                if (compilerOptions.removeComments) {
+                    // removeComments is true, only reserve pinned comment at the top of file
+                    // For example:
+                    //      /*! Pinned Comment */

```

Comment:
This seems wrong (and seems like a very strange thing to explain to people using the product).  "We will preserve pinned comments.  But only at the very top of a file".

It also seems like we'll be breaking people who use pinned comments today (a feature we added precisely for people that wanted to use --removeComments, but who wanted some comments to remain).